### PR TITLE
[IMP] account: bank statement misc improvements.

### PR DESCRIPTION
**Pull request by vin-odoo**
In order to better display the functionalities, we will change how the demo data for bank statements are created. There will now be one bank statement and multiple bank transactions not linked to a statement. Also rewords one error to be clearer.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/demo/account_demo.py [hunk 1]
```diff
@@ -215,11 +220,11 @@ def _get_demo_data_attachment(self):
         cid = self.env.company.id
         ref = self.env.ref
         return ('ir.attachment', {
-            f'{cid}_ir_attachment_bank_statement_1': {
+            f'{cid}_ir_attachment_bank_statement_0': {
```
Comment (william-andre) [inline addons/account/demo/account_demo.py:223]: You need to update this file too, it contains all the other lines too

Comment (vin-odoo) [reply addons/account/demo/account_demo.py:223]: I've updated the pdf :)

Comment (william-andre) [reply addons/account/demo/account_demo.py:223]: Actually you can remove the file
It needs to be "posted" on all the lines, but the feature is not implemented yet (when you generate the statement it should be in the chatter of all the lines)

Comment (vin-odoo) [reply addons/account/demo/account_demo.py]: Done, removed the PDF, the ir.attachment, and the message that posted it.

**Review by william-andre — APPROVED** [review]
@robodoo r+

File: addons/account/models/account_bank_statement.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def default_get(self, fields_list):
                 .filtered(lambda line: not line.statement_complete) \
                 .sorted()
             if not lines:
-                raise UserError(_('No editable line selected.'))
+                raise UserError(_('One or more selected lines already belong to a complete statement.'))
```
Comment (poma-odoo) [inline addons/account/models/account_bank_statement.py:61]: This error message is now even more misleading.
It should be *All selected lines are already connected to a complete statement*

Comment (vin-odoo) [reply addons/account/models/account_bank_statement.py:61]: Benj gave me this line, I didn't look much more in the context of it (should have). Will you correct it in one of your branches?

Comment (william-andre) [reply addons/account/models/account_bank_statement.py:61]: Well it isn't wrong :sweat_smile: 
"One or more" includes "all" :upside_down_face:

Comment (william-andre) [reply addons/account/models/account_bank_statement.py:61]: > This error message is now even more misleading.

I don't agree though. The previous comment really didn't mean anything unless you wrote the module. This one is just not accurate, but understandable for a regular user.

Comment (poma-odoo) [reply addons/account/models/account_bank_statement.py:61]: @william-andre  You're right, somehow, but the previous error was not giving any information at all, so it was less misleading :D 
I had actually changed it, but it came back through one of the many revert/rebase/patches.
@vin-odoo Sure, will do :)

**Pull request closed without merging**